### PR TITLE
Impro 1121 rename probability cubes

### PR DESCRIPTION
**Pull request by markysparks**
Addresses #GitHubissuenum

Rename all IMPROVER cubes that contain probabilities like `probability_of_rainfall_rate` to follow a specified convention that includes whether the probability is above or below a threshold within the cube name, so that the cube name better describes the contents of the cube. For example:

-  `probability_of_rainfall_rate_above_threshold`
-  `probability_of_visibility_in_air_below_threshold`
-  `probability_of_lwe_precipitation_rate_in_vicinity_above_threshold`

Testing:
 - [x] Ran tests and they passed OK
 - [x] Update/added new tests for the changes

This PR covers an initial set of changes with a final set being made in PR #817 (this PR will be rebased on master once these initial changes have been merged in).

### CODE REVIEW

File: bin/improver-threshold [hunk 1]
```diff
@@ -162,8 +162,17 @@ def main():
         # smooth thresholded occurrences over local vicinity
         result_no_collapse_coord = OccurrenceWithinVicinity(
             args.vicinity).process(result_no_collapse_coord)
-        result_no_collapse_coord.rename(
-            result_no_collapse_coord.name() + '_in_vicinity')
+        cube_name = result_no_collapse_coord.name()
```
Comment (fionaRust) [inline bin/improver-threshold]: This could be a function that we move somewhere and unit test. What do you think?

Comment (markysparks) [reply bin/improver-threshold]: This CLI already has quite a bit of 'pre-processing' going on and I didn't think too much of adding this change here. However I shouldn't use this as an excuse and I agree that processing should be kept outside of CLI's as far as possible.

So I've moved this into a function within` utilities/cube_metadata.py ` as this seemed the most logical place to put it. Accompanying unit test created.

File: lib/improver/tests/utilities/test_cube_metadata.py [hunk 2]
```diff
@@ -612,6 +613,29 @@ def test_warnings_on_works(self, warning_list=None):
         self.assertEqual(result.attributes['new_attribute'],
                          'new_value')
 
+    def test_in_vicinity_name_format(self):
```
Comment (fionaRust) [inline lib/improver/tests/utilities/test_cube_metadata.py]: This test needs to be in a new class, as we have one test class for each function and at the moment it is in a class for a different function.
I also expect we need more than one test, we need to test with and without vicinity, and probably some edge cases

Comment (markysparks) [reply lib/improver/tests/utilities/test_cube_metadata.py]: Test placed in new class. Test has been split up to separately test renaming and case of `_above/below_threshold` not present. Without `in_vicinity` is already tested (this would be the normal case) in the first test. 

I could add a test for` in_vicinity` already present - the result from the function would be e.g `probability_of_X_rate_in_vicinity_in_vicinity_above_threshold` . Do you think this is a good idea? Perhaps it would be better to raise an exception from the function in that case and test for the exception being raised?

File: lib/improver/tests/utilities/test_cube_metadata.py [hunk 3]
```diff
@@ -804,5 +805,44 @@ def test_history_append_no_existing(self):
         self.assertTrue("Nowcast" in cube.attributes["history"])
 
 
+class Test_probability_cube_in_vicinity_rename(IrisTest):
```
[the remark this answers is not in the document]
Comment (markysparks) [reply lib/improver/tests/utilities/test_cube_metadata.py]: Yes thanks - I should have realised that! I've accepted your change...
